src/rotator_library/http_client_pool.py

```
@@ -0,0 +1,514 @@
+            return self._streaming_client
+        else:
+            client = self._non_streaming_client
+            if self._is_client_closed(client):
+                lib_logger.warning(
+                    "Non-streaming HTTP client was closed, recreating..."
+                )
+                self._non_streaming_client = await self._create_client(streaming=False)
+                self._stats["reconnects"] += 1
+            return self._non_streaming_client
+
+    def get_client(self, streaming: bool = False) -> httpx.AsyncClient:
```

Ruff (0.15.4)

[error] Boolean-typed positional argument in function definition

(FBT001)